common/src/sweights.cpp

```diff
@@ -0,0 +1,147 @@
+		int ncontr = Config::ncontr;
+		int nvar_md = Config::nvar_md;
+		int nvar_mb = Config::nvar_mb;
+		double frac[ncontr];
+		for (int i=0; i<ncontr; i++){
+			frac[i] = std::abs(res[ncontr*(nvar_md+nvar_mb)+i]);
+			std::cout << frac[i] << " frac " << i << std::endl;
+		}
+
+		TChain ch(Config::chainName.c_str());
+
+		double Tau, D_M, mu_PT, mu_P, mu_eta, K_PT, B_M, missPT;
+  		bool charge;
+
+		ch.Add(path_data.c_str());
+		std::vector<std::pair<double, double>> vect_2D={};
+   		ch.SetBranchAddress("Tau", &Tau);
+   		ch.SetBranchAddress("B_M", &B_M);
+   		ch.SetBranchAddress("missPT", &missPT);
+   		ch.SetBranchAddress("D_M", &D_M);
+   		ch.SetBranchAddress("mu_PT", &mu_PT);
+   		ch.SetBranchAddress("mu_P", &mu_P);
+   		ch.SetBranchAddress("mu_eta", &mu_eta);
+   		ch.SetBranchAddress("K_PT", &K_PT);
+   		ch.SetBranchAddress("truecharge", &charge);
+   		//ch.SetBranchAddress("blinded_charge", &bcharge);
+
+		std::vector<double> vect_time = {};
+		std::cout << " tuuuuuu\n";
+
+		int Nevents = Config::nentries;
+		if (Nevents==-1)
+			Nevents = ch.GetEntries();
+		for (int i=0; i<Nevents;i++){
+      			ch.GetEntry(i);
+			if (mu_PT < Config::muPTmin || mu_P < Config::muPmin || mu_eta < Config::eta_min || mu_eta > Config::eta_max)
+                                continue;
+      			double B_MMcorr = B_M +2.0*missPT;
+      			if (B_MMcorr < Config::minBMcorr || B_MMcorr > Config::maxBMcorr)
+                        continue;
+
+                	if (D_M < Config::minDM || D_M > Config::maxDM) continue;
+			if (Tau<Config::tMin || Tau>Config::tMax) continue; 
+      			if (charge==BBbar){
+				vect_2D.push_back(std::make_pair(D_M, B_MMcorr));
+				vect_time.push_back(Tau);
+			}
+			
+      		}
+		std::cout << vect_2D.size() << " siiiizieeeee \n";
+
+		TMatrixD matrixV(ncontr, ncontr);
+		const auto& B_PDFs = Config::getVectorPDFs("Bmass");
+		const auto& D_PDFs = Config::getVectorPDFs("Dmass");
+		double nevents = (double) vect_2D.size();
+		for (int i=0; i<ncontr; i++){
+			D_PDFs[i]->CalcIntegral(&res[i * Config::nvar_md], Config::minDM, Config::maxDM);
+                        B_PDFs[i]->CalcIntegral(&res[Config::ncontr * Config::nvar_md + i * Config::nvar_mb], Config::minBMcorr, Config::maxBMcorr);
+		}
+		for (int n=0; n<ncontr; n++){
+			for(int j=0; j<ncontr; j++){
+				matrixV (n, j)=0.0;
+				int emax = (int)vect_2D.size();
+	        		for (int e=0; e<emax; e++){
+					double mdass = std::get<0>(vect_2D[e]);
+                        		double mcorr = std::get<1>(vect_2D[e]);
+
+					double sum_k = 0.0;
+					double md_like, mb_like, md_like2, mb_like2;
+		    			for (int i=0; i<ncontr; i++){
+						md_like = D_PDFs[i]->EvalPDF(&mdass, &res[i * Config::nvar_md]);
+                	                	mb_like = B_PDFs[i]->EvalPDF(&mcorr, &res[Config::ncontr * Config::nvar_md + i * Config::nvar_mb]);
+				//std::cout << md_like << "   " << mb_like << "  " << nevents << std::endl;
+						sum_k += md_like*mb_like*frac[i]*nevents;	
```

Comment:
Wydaje mi się że ta część nie zależy ani od `n` ani od `j` do tego jest taka sama jak w linii 127.

Przepiszę to tak że jest liczone raz na początku. OK?